src/ralph/lib/polymorphic/models.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+"""
+Polymorphic using inherited models in Django.
+Displays information in queryset list about the model that inherits
+the polymorphic model
+
+Example:
+    >>>  Model.polymorphic_objects.all()
+    [
+        <Model1: model1: test>,
+        <Model2: model2: test>,
+        <Model3: model3L test>
+    ]
+"""
+from itertools import groupby
+
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+
+class PolymorphicQuerySet(models.QuerySet):
+
+    def iterator(self):
+        """
+        Override iterator:
+            - Iterate for all objects and collected ID
+            - For each ContentType generates additional queryset
+            - Returns iterator with different models
+        """
+        result = []
+        content_type_model_map = {}
+        for obj in super().iterator():
+            content_type_model_map[
+                obj.content_type_id
+            ] = obj.content_type.model_class()
+            result.append((
+                obj.content_type_id, obj.pk)
+            )
+        result = groupby(result, lambda x: x[0])
+
+        result_query = []
+        for k, v in result:
+            model = content_type_model_map[k]
+            polymorphic_models = getattr(model, '_polymorphic_models', [])
+            if polymorphic_models and model not in polymorphic_models:
+                result_query.extend(
+                    model.objects.filter(pk__in=[i[1] for i in v])
+                )
+
+        for obj in result_query:
+            yield obj
+
+
+class PolymorphicBase(models.base.ModelBase):
```

Comment:
please add some comment what is going on here (how newly created class is affected and base classes)